### PR TITLE
message 功能进行实现 Issue#26

### PR DESCRIPTION
**Pull request by guo40020**
#26

### CODE REVIEW

File: src/util/vcard-parser.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+export interface IVcard {
```
Comment (su-chang) [inline src/util/vcard-parser.ts:1]: Please merge `dev` branch to this branch, and then we could move this util function to the directory `pure-function-helpers`.

File: src/puppet-whatsapp.ts [hunk 2]
```diff
@@ -414,71 +415,182 @@ class PuppetWhatsapp extends PUPPET.Puppet {
...
+   */
   override async messageMiniProgram (messageId: string): Promise<PUPPET.MiniProgramPayload> {
     log.verbose('PuppetWhatsApp', 'messageMiniProgram(%s)', messageId)
-    // const attachment = this.mocker.MockMessage.loadAttachment(messageId)
-    // if (attachment instanceof MiniProgram) {
-    //   return attachment.payload
-    // }
-    return {
-      title : 'mock title for ' + messageId,
+    PUPPET.throwUnsupportedError()
```
Comment (bung87) [inline src/puppet-whatsapp.ts:533]: should return PUPPET.throwUnsupportedError() which will throw error

File: src/puppet-whatsapp.ts [hunk 3]
```diff
@@ -414,71 +415,182 @@ class PuppetWhatsapp extends PUPPET.Puppet {
...
+  }
+
+  override async messageSendMiniProgram (conversationId: string, miniProgramPayload: PUPPET.MiniProgramPayload): Promise<void> {
+    log.verbose(
+      'PuppetWhatsApp',
+      'messageSendMiniProgram(%s, %s)',
+      conversationId,
+      JSON.stringify(miniProgramPayload),
+    )
+    PUPPET.throwUnsupportedError()
```
Comment (bung87) [inline src/puppet-whatsapp.ts:586]: should return PUPPET.throwUnsupportedError() which will throw error

**Review by su-chang — DISMISSED** [review]

**Comment by su-chang** [comment]
Please resolve conflict and then merge this PR.

**Review by bung87 — APPROVED** [review]

**Pull request merged**